test/e2e/auth/projected_podcertificate.go

```diff
@@ -0,0 +1,549 @@
+				if err != nil {
+					framework.Failf("Pod failed to start: %v", err)
+				}
+
+				ginkgo.By("Fetching certificate and verifying duration...")
+				logs, err := e2epod.GetPodLogs(ctx, f.ClientSet, namespace, createdPod.Name, createdPod.Spec.Containers[0].Name)
+				if err != nil {
+					framework.Failf("failed to get pod logs: %v", err)
+				}
+				certDER := extractCertDERFromLogs(logs)
+				if certDER == nil {
+					framework.Failf("Could not find CERTIFICATE block in pod logs:\n%s", logs)
+				}
+				certs, err := x509.ParseCertificates(certDER)
+				if err != nil {
+					framework.Failf("failed to parse certificate: %v", err)
+				}
+				gomega.Expect(certs).ToNot(gomega.BeEmpty())
+				cert := certs[0]
+				lifeTime := cert.NotAfter.Sub(cert.NotBefore)
+				expectedDuration := time.Duration(requestedSeconds) * time.Second
+				ginkgo.By(fmt.Sprintf("Verifying certificate duration %v is close to %v", lifeTime, expectedDuration))
+				gomega.Expect(lifeTime).To(gomega.BeNumerically("==", expectedDuration), "Certificate duration should be 1 hour")
+			})
+
+			ginkgo.It("should fail pod startup when MaxExpirationSeconds exceeds maximum (91d)", func(ctx context.Context) {
+				namespace := f.Namespace.Name
+				ginkgo.By("Using namespace: " + namespace)
+
+				// Exceeds 91 days
+				tooLongSeconds := int32((91 * 24 * 60 * 60) + 1)
+				testPod := createMaxExpirationTestPod(namespace, "too-long-duration-pod", &tooLongSeconds)
+				ginkgo.By("Creating pod requesting >91d MaxExpirationSeconds...")
+				_, err := f.ClientSet.CoreV1().Pods(namespace).Create(ctx, testPod, metav1.CreateOptions{})
+				if err == nil {
+					framework.Fail("Error message does not match the expected.")
+				}
+				if err != nil {
+					if !strings.Contains(err.Error(), "if provided, maxExpirationSeconds must be <= 7862400") {
+						framework.Failf("failed to create pod: %v", err)
+					}
+				}
+			})
+
+			ginkgo.It("should fail pod startup when MaxExpirationSeconds is less than minimum (1h)", func(ctx context.Context) {
+				namespace := f.Namespace.Name
+				ginkgo.By("Using namespace: " + namespace)
+
+				// Less than 1 hour
+				tooShortSeconds := int32(3599)
+				testPod := createMaxExpirationTestPod(namespace, "too-short-duration-pod", &tooShortSeconds)
+				ginkgo.By("Creating pod requesting <1h MaxExpirationSeconds...")
+				_, err := f.ClientSet.CoreV1().Pods(namespace).Create(ctx, testPod, metav1.CreateOptions{})
+				if err != nil {
+					if !strings.Contains(err.Error(), "if provided, maxExpirationSeconds must be >= 3600") {
+						framework.Fail("Error message does not match the expected")
+					}
+				}
+			})
+		})
+	})
+
+// createServerObjects creates the Deployment and Service objects for the mTLS server.
+func createServerObjects(namespace string) (*appsv1.Deployment, *v1.Service) {
+	replicas := int32(1)
+	serverLabels := map[string]string{"app": "server"}
+	signerNameVar := spiffeSignerName
+	serverDeployment := &appsv1.Deployment{
```

Comment:
I think perhaps using the fixtures might be a bit easier on the reader: https://github.com/kubernetes/kubernetes/blob/e1773f1afd16d7e2d9e0cd9b2c853734abb76ea5/test/e2e/framework/deployment/fixtures.go#L42